### DISCUSSION TITLE
Storybook 7: How to render multiple components inside a single <Canvas/>

### DISCUSSION THREAD

**Discussion by hieucd04** [storybookjs/storybook#22587] [Help]
In storybook 6, I can do this:

```
<Canvas withSource={SourceState.CLOSED}>
    <Story name="Sizes">
        <Label style={Variant.Small}>Small</Label>
        <Label style={Variant.Medium}>Medium</Label>
        <Label style={Variant.Large}>Large</Label>
    </Story>
</Canvas>
```

![image](https://github.com/storybookjs/storybook/assets/89077642/03f0b6e0-f791-4257-bc44-d805a4bc8df3)

What I did was that I put 3 `<Label>` components inside a single `<Canvas>`.

---
In storybook 7, the recommended way is to use `of` props. But the `of` props doesn't accept array. It only accepts a single story.
So, in storybook 7, how do I put multiple components inside a single `<Canvas>` ?

**Comment by sgodard** [comment]
In my mdx file, I'm doing this

```
<Canvas>
  <Story of={AlertStories.SeverityInfo} />
  <Story of={AlertStories.SeveritySuccess} />
  <Story of={AlertStories.SeverityWarning} />
  <Story of={AlertStories.SeverityError} />
</Canvas>
```

**Reply by zorochimaru777** [reply]
But a warning will appear with this too.

**Reply by VBikashR** [reply]
got any solution?

**Comment by domyen** [comment]
Thanks for the question! Our recommendation is to have a story that renders multiple components. 

(Note: this story will appear in the sidebar though)

**Reply by alexastef** [reply]
when i use stories this way, my MDX file only renders the first component. can you share an example with multiple variants that are all rendered in an MDX file? I'm not sure what I'm doing wrong. i just made [this ticket](https://github.com/storybookjs/storybook/issues/25495) with more detail
